packages/cli/lib/cli/cliStates.js

```diff
@@ -102,6 +102,52 @@ module.exports = class CLIStates extends CLICommand {
         });
     }
 
+    /**
+     * Checks if state is a binary state
+     * @param {string} id id of the state
+     * @param {object} objects the objects db
+     * @param {ioBroker.OtherObject?} obj cached object
+     * @return {Promise<boolean>}
+     * @private
+     */
+    async _isBinary(id, objects, obj) {
+        obj = obj || (await objects.getObjectAsync(id));
+
+        if (obj && (obj.binary || (obj.common && obj.common.type === 'file'))) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     * Get and show binary state
+     *
+     * @param {string} id if of the state
+     * @param {object} states states db
+     * @return Promise<void>
+     * @private
+     */
+    async _getBinary(id, states) {
+        try {
+            /** @type Buffer | null */
+            const state = await states.getBinaryState(id);
+
+            if (!state) {
+                CLI.error.stateNotFound(id);
+                return;
+            }
+
+            if (Buffer.isBuffer(state)) {
+                console.log(state.toString('utf-8'));
```

Comment:
utf-8, maybe this should be user specified.. as mentioned, which then would be a pro for a `state getbinary` cli command

Reply:
I do not like this "mix" of states and binary states in one CLI ... I would really like a "get-binary" with the state id and an optional secodn parameter for the "output encoding" (default is "binary") and allowed is utf-8, ascii, hex and base64